src/app/stream-deck/interfaces.ts

```diff
@@ -0,0 +1,26 @@
+import { InventoryBucket } from 'app/inventory/inventory-buckets';
+
+export interface StreamDeckMessage {
```

Comment:
Try using a discriminated union in TypeScript instead. For example:

```ts
interface SearchAction {
  action: 'search';
  search: string;
}

interface LoadoutAction {
  action: 'loadout';
  loadout: string;
  character: string;
}

type Actions = SearchAction | LoadoutAction;
```

This way TypeScript will be able to tell which properties are valid for which message based on the `action` field.